### PR TITLE
README improvements mainly for contributors

### PR DESCRIPTION
**Pull request by tbouffard**
For users
  - Bonita version support
  - branch or tags to use

For contributors
  - new developing dedicated section
  - add new Bonita version support
  - create tag and release

### CODE REVIEW

**Comment by tbouffard** [comment]
@dcoille I have tested the tag and release doc to manage the [7.10.1-1 release](https://github.com/Bonitasoft-Community/Build-Bonita/releases/tag/7.10.1-1)
I hope this will help you for Bonita 7.10.2

File: README.md [hunk 1]
```diff
@@ -44,31 +41,147 @@ rm -rf ~/.gradle/caches
...
+  - merge `dev` into `bonita_7.10.0_GA`
+  - create a Pull Request from `bonita_7.10.0_GA` targeting the `master` branch
+  - merge the PR into `master` when the build passed (eventually after fixing any issues related to the merge)
+
+
+## New Release
+
+Release are cut when
+- a new Bonita version (GA or maintenance) is supported by `Build-Bonita`
+- significant improvements have been made to t
```
Comment (amottier) [inline README.md]: Little typo here

Comment (tbouffard) [reply README.md]: yes indeed 😈 . I am on it

Comment (tbouffard) [inline README.md]:
```suggestion
- significant improvements have been made in the `Build-Bonita` build script for the latest supported Bonita version
```

Comment (tbouffard) [reply README.md]:
```suggestion
- significant improvements have been made in the `Build-Bonita` build script for the latest supported Bonita version
```

**Comment by amottier** [comment]
LGTM

**Comment by tbouffard** [comment]
@amottier the PR seems now ok for you, could you please change your review status? I can also dismiss your review 😈 
@dcoille is it ok for you, or do you want to give it a try to manage the 7.10.2 release?

File: README.md [hunk 2]
```diff
@@ -44,31 +41,147 @@ rm -rf ~/.gradle/caches
...
+- branch and pull request creation
+
+See [GitHub help](https://help.github.com/en/github/managing-files-in-a-repository/editing-files-in-your-repository) for
+more details
+
+### Bonita maintenance version
+
+- create a new branch starting from the `master` branch, for instance `maintenance_7.10.2`
+- update the `build-script.sh` file and update the `BONITA_VERSION` variable
+- create a Pull Request targeting the `master` branch (TODO link to github doc)
```
Comment (tbouffard) [inline README.md]:
```suggestion
- [create a Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request) targeting the `master`
```

File: README.md [hunk 3]
```diff
@@ -44,31 +41,147 @@ rm -rf ~/.gradle/caches
...
+- wait for build to pass, this should work without any other modifications
+- merge the Pull Request after successful build
+
+
+### Bonita development version
+
+- create a new branch starting from the `dev` branch, for instance `dev_7.11.0.W10`
+- update the `build-script.sh` file and update the `BONITA_VERSION` variable
+- create a Pull Request targeting the `dev` branch
```
Comment (tbouffard) [inline README.md]:
```suggestion
- [create a Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request) targeting the `dev` branch
```

File: README.md [hunk 4]
```diff
@@ -44,31 +41,147 @@ rm -rf ~/.gradle/caches
...
+See [GitHub help](https://help.github.com/en/github/managing-files-in-a-repository/editing-files-in-your-repository) for
+more details
+
+### Bonita maintenance version
+
+- create a new branch starting from the `master` branch, for instance `maintenance_7.10.2`
+- update the `build-script.sh` file and update the `BONITA_VERSION` variable
+- create a Pull Request targeting the `master` branch (TODO link to github doc)
+- wait for build to pass, this should work without any other modifications
+- merge the Pull Request after successful build
```
Comment (tbouffard) [inline README.md]:
```suggestion
- [merge the Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/merging-a-pull-request) after successful build
```

File: README.md [hunk 5]
```diff
@@ -44,31 +41,147 @@ rm -rf ~/.gradle/caches
...
+
+- create a new branch starting from the `dev` branch, for instance `dev_7.11.0.W10`
+- update the `build-script.sh` file and update the `BONITA_VERSION` variable
+- create a Pull Request targeting the `dev` branch
+- wait for build to run
+- failures often happen because of new components to be added or removed, build options of some components to be updated
+  - try to fix, then commit and iterate until build pass
+  - see [#32](https://github.com/Bonitasoft-Community/Build-Bonita/pull/32) or
+  [#48](https://github.com/Bonitasoft-Community/Build-Bonita/pull/48) for instance
+- merge the Pull Request after successful build
```
Comment (tbouffard) [inline README.md]:
```suggestion
- [merge the Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/merging-a-pull-request) after successful build
```

File: README.md [hunk 6]
```diff
@@ -44,31 +41,147 @@ rm -rf ~/.gradle/caches
...
+contributors
+- `master` --> `dev`: all the time, especially after adding support for a new maintenance version. Allow to get new
+improvements applied to maintenance versions, avoid subsequent merge conflicts, ...
+- `dev` --> `master`: on GA release, `master` is going to become the maintenance branch for the new Bonita released
+version. It is highly advised to do the merge in a dedicated branch as some issue occurred at that stage in the past
+  - first, ensure that `master` has been merged into`dev`
+  - create a new `bonita_7.10.0_GA` branch starting from `master` branch
+  - merge `dev` into `bonita_7.10.0_GA`
+  - create a Pull Request from `bonita_7.10.0_GA` targeting the `master` branch
+  - merge the PR into `master` when the build passed (eventually after fixing any issues related to the merge)
```
Comment (tbouffard) [inline README.md]:
```suggestion
  - [merge the Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/merging-a-pull-request) into `master` when the build passed (eventually after fixing any issues related to the merge)
```

File: README.md [hunk 7]
```diff
@@ -44,31 +41,147 @@ rm -rf ~/.gradle/caches
...
+Follow the same lifecycle as Bonita component repositories. Merge are currently done manually by `Build-Bonita`
+contributors
+- `master` --> `dev`: all the time, especially after adding support for a new maintenance version. Allow to get new
+improvements applied to maintenance versions, avoid subsequent merge conflicts, ...
+- `dev` --> `master`: on GA release, `master` is going to become the maintenance branch for the new Bonita released
+version. It is highly advised to do the merge in a dedicated branch as some issue occurred at that stage in the past
+  - first, ensure that `master` has been merged into`dev`
+  - create a new `bonita_7.10.0_GA` branch starting from `master` branch
+  - merge `dev` into `bonita_7.10.0_GA`
+  - create a Pull Request from `bonita_7.10.0_GA` targeting the `master` branch
```
Comment (tbouffard) [inline README.md]:
```suggestion
  - [create a Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request) from `bonita_7.10.0_GA` targeting the `master` branch
```

**Comment by tbouffard** [comment]
> Maybe only one question: why creating the tag locally instead of doing it on gitHub as we create the release?

I didn't find the documentation and wasn't sure how it really works but finally, I have just got it: https://help.github.com/en/github/administering-a-repository/managing-releases-in-a-repository#creating-a-release
Ok, here is what I am going to do
- document how to create the tag directly when publishing the release
- keep the local command if push occured on the master branch that are not related to the release

**Pull request merged**